rpcclient/infrastructure.go

```diff
@@ -947,7 +963,7 @@ func (c *Client) sendCmd(ctx context.Context, cmd interface{}) chan *response {
 func (c *Client) sendCmdAndWait(ctx context.Context, cmd interface{}) (interface{}, error) {
 	// Marshal the command to JSON-RPC, send it to the connected server, and
 	// wait for a response on the returned channel.
-	return receiveFuture(c.sendCmd(ctx, cmd))
+	return receiveFuture(ctx, c.sendCmd(ctx, cmd).c)
```

Comment:
The entire `sendCmdAndWait` method should be removed now, as it's no longer needed for anything.  The fact the updated code needs to stutter like this is a pretty good tip off of that fact!